src/Integration/LocalServices/SourceControlledFileSystem.cs

```diff
@@ -146,7 +152,10 @@ void IFileSystem.CreateDirectory(string path)
         #region Helpers
         private bool CheckoutForEdit(params string[] fileNames)
         {
-            VsQueryEditFlags flags = VsQueryEditFlags.SilentMode | VsQueryEditFlags.DetectAnyChangedFile;
+            VsQueryEditFlags flags = VsQueryEditFlags.SilentMode | VsQueryEditFlags.DetectAnyChangedFile |
+                // Force no prompting here to fix SLVS#801 (otherwise TFS would prompt about the slconfig
+                // and generated ruleset files in new connected mode that are not in the solution)
+                VsQueryEditFlags.ForceEdit_NoPrompting;
```

Comment:
This is the bug fix - passing an extra flag.

I've manually tested the change in VS2017 on solution that are not in SCC, in Git and in TfVC.